plugins/module_utils/network/sonic/config/lag_interfaces/lag_interfaces.py

```diff
@@ -372,6 +435,72 @@ def create_lag_interfaces_requests(self, commands):
                 requests.append(request)
         return requests
 
+    def create_ethernet_segment_requests(self, diff_members, have):
+        es_commands = []
+        es_requests = []
+        es_path = 'data/openconfig-network-instance:network-instances/network-instance=default/evpn/ethernet-segments'
+        df_pref_path = es_path + '/ethernet-segment=%s/df-election/config/preference'
+
+        for cmd in diff_members:
+            po_name = cmd['name']
+            cmd_es = cmd.get('ethernet_segment', None)
+            if cmd_es:
+                es = dict()
+                have_po = next((po for po in have if po['name'] == po_name), {})
+                have_es = have_po.get('ethernet_segment', {})
+
+                if cmd_es.get('esi_type'):
```

Comment:
It is needed due to "diff" operation removes it from cmd.

Reply:
I can see why this is needed for correct handling of 'overridden' and 'replaced' state because the previously configured 'esi_type' value may have been deleted.

Resolved.